### PR TITLE
[FIX] search with computed fields in path

### PR DESCRIPTION
**Pull request by hbrunn**
cleaned up version of https://github.com/odoo/odoo/pull/5898

### CODE REVIEW

**Comment by hbrunn** [comment]
so the first issue is that nonstored computed fields don't have a column (https://github.com/odoo/odoo/blob/8.0/openerp/fields.py#L695) which is why we need to check for that, and then we need to walk through the dotted path if it includes a x2x field

File: openerp/osv/expression.py [hunk 1]
```diff
@@ -858,22 +858,22 @@ def push_result(leaf):
...
             # Making search easier when there is a left operand as column.o2m or column.m2m
-            elif len(path) > 1 and column._type in ['many2many', 'one2many']:
+            elif len(path) > 1 and column and column._type in ['many2many', 'one2many']:
                 right_ids = comodel.search(cr, uid, [(path[1], operator, right)], context=context)
                 table_ids = model.search(cr, uid, [(path[0], 'in', right_ids)], context=dict(context, active_test=False))
                 leaf.leaf = ('id', 'in', table_ids)
                 push(leaf)
 
-            elif not column:
+            elif not column or field and field.search:
```
Comment (rco-odoo) [inline openerp/osv/expression.py]: @hbrunn that extra condition is not useful; `field` always exists, and there is a test on `field.search` below.

**Comment by rco-odoo** [comment]
@hbrunn can you please modify the code according to my comment?
Then please rebase your branch, and make sure the tests pass.
Otherwise the proposal looks good to me.
Thanks!

**Comment by hbrunn** [comment]
@rco-odoo thanks, done

**Pull request merged**